sycl/test-e2e/InvokeSimd/Regression/nbarrier_exec_in_order.cpp

```diff
@@ -127,27 +134,53 @@ bool test(QueueTY q) {
     out[i] = -1;
   }
 
+  auto dev = q.get_device();
+  auto deviceSLMSize =
+      dev.template get_info<sycl::info::device::local_mem_size>();
+  if constexpr (UseSLM)
+    std::cout << "Local Memory Size: " << deviceSLMSize << std::endl;
```

Comment:
why only this printing is guarded by UseSLM check?

Reply:
+1, i would have expected the SLM error below to also be guarded.

Reply:
I decided to move this print outside of the _test_ function since the amount of available memory will never change between cases. I also added a guard to each case that uses SLM.